### ISSUE TITLE
LG Signature Climate complex lsa50a

### ISSUE THREAD

**Issue by mo1seev** [ollo69/ha-smartthinq-sensors#288]
Hello!

Basic functions (power/pmi sensors) works perfect but it would be nice to have more functions as:
- Speed (/Auto/High/Mid/Low)
- Mode (Silent/Purify/Humidi)
- Filters status
- Lighting mode and state
- Ionizer mode and state

[smartthinq_sensors-d7b0c5028506fd43b7a51fffce138b9c-ÐÑÐ¸ÑÑÐ¸ÑÐµÐ»Ñ Ð²Ð¾Ð·Ð´ÑÑÐ°-b2297ec6b47ed56c1495202e19e5d576.json.txt](https://github.com/ollo69/ha-smartthinq-sensors/files/8010065/smartthinq_sensors-d7b0c5028506fd43b7a51fffce138b9c-D.N.D.N.N.D.N.D.D.N.D.D.D.D.N.N.D.-b2297ec6b47ed56c1495202e19e5d576.json.txt)

**Comment by ollo69** [comment]
Could you clarify if and how the Mode affect the Speed? Could also clarify what is Ioniner (is alternative to others modes)

**Comment by mo1seev** [comment]
In silent mode speed choice is disabled (i think its forced to low). All speeds are available in humidi and purify modes.

Ionizer is an additional option (not separate mode) and work with any modes.

**Comment by ollo69** [comment]
Ok, I will try to work on this next w.e.

**Comment by mo1seev** [comment]
Awesome news! Thx! And one more - this device also have humidi sensor. Can you add him too?

**Comment by ollo69** [comment]
This is partial implemented in `v0.23.0`:

- Speed (/Auto/High/Mid/Low) -> Based on Fan entities capabilities
- Filters status sensors
- Humidity sensor

For the `Mode`, it is more complex in HA. It should be probably implemented as switchs, keeping in mind the interaction with Fan modes. I'll think about this.

Please provide feed-back about current implemented functionality.

**Comment by mo1seev** [comment]
@ollo69 all work perfectly!

<img width="342" alt="image" src="https://user-images.githubusercontent.com/23101489/172105019-900f60fa-b54a-425f-866c-7cb3d32b4c83.png">

Also i find one more air quality sensor: odor. I think its airState.quality.odor

**Comment by ollo69** [comment]
What about Fan control? Did you test it? Does it work?


> Also i find one more air quality sensor: odor. I think its airState.quality.odor. It would be nice to add it too if it's not a big job.

What does this sensors means (unit, possible values, etc)

**Comment by mo1seev** [comment]
> What about Fan control? Did you test it? Does it work?

The fan speed is changed via the slider, not via the drop-down menu. Is that the plan?
<img width="383" alt="image" src="https://user-images.githubusercontent.com/23101489/172202495-51c78739-1aab-426a-af4e-022a9abe2572.png">

In drop-down menu only Auto:
<img width="397" alt="image" src="https://user-images.githubusercontent.com/23101489/172202893-a1fe8676-4cda-4aa7-ad4a-14ae52c9e5c6.png">

**Comment by mo1seev** [comment]
>What does this sensors means (unit, possible values, etc)

i think its OU/m3

**Comment by ollo69** [comment]
> The fan speed is changed via the slider, not via the drop-down menu. Is that the plan?

Yes, by design FAN speed must be represented as percentage. Preset modes can be used for specific fan modes (as Auto), and must be reset when a specific FAN speed percentage is set. You can see more details [here](https://developers.home-assistant.io/docs/core/entity/fan).

Does speed change and Auto mode works properly?

**Comment by ollo69** [comment]
> i think its OU/m3

But what are the values range for this sensor? What is the data type (float, int...)

**Comment by mo1seev** [comment]
> Preset modes can be used for specific fan modes (as Auto), and must be reset when a specific FAN speed percentage is set.

So maybe we can use only `preset_modes`? I, as a user, do not know what percentage speed value corresponds to preset_modes and it will be difficult for me to set the correct value using the slider.

> But what are the values range for this sensor? What is the data type (float, int...)

I think its same values as PM* sensors

**Comment by ollo69** [comment]
> So maybe we can use only `preset_modes`? I, as a user, do not know what percentage speed value corresponds to preset_modes and it will be difficult for me to set the correct value using the slider.

No, this is not how it should be. From dev docs:

```
Preset modes should not include named (manual) speed settings as these should be represented as percentages.
```

When you set a percentage, I write an additional attribute that will show to you the current named mode.

**Comment by ollo69** [comment]
> And should we use the same logic to select the operating modes?

Again no, because you can have an op mode selected (Clean or Humidity) and at the same time change the Fan speed, and this is not compliant with specification:

```
Manually setting a speed must disable any set preset mode. If it is possible to set a percentage speed manually 
without disabling the preset mode, create a switch or service to represent the mode
```

So `OpMode` should be managed with some switches or a dedicated service

The real problems is that in HA do not exist a dedicated entity for `air-purifier`, so we must use other entities and adapt for this use. **But entities specification must be respected!**

**Comment by mo1seev** [comment]
> So OpMode should be managed with some switches or a dedicated service

Maybe as you realise it on AC?

<img width="387" alt="image" src="https://user-images.githubusercontent.com/23101489/172243937-66dc8a88-e3b2-4669-80ef-e05dd1e01545.png">

**Comment by ollo69** [comment]
AC must support temperature (current and target).

**Comment by mo1seev** [comment]
@ollo69 hello! I find bug. After switching off in hassio, the device physically turns off. But his status in hassio remains on. I rollback to 0.22.0 for test - it work correctly.

**Comment by mo1seev** [comment]
@ollo69 hello! I want to come back to this issue. Can you add control for Lighting mode and state for this device? Bcs when its turned on, green light always on. And no way to disable this. Light control support will help create automation that turns off the bright green light when the device is turned on.

**Comment by ollo69** [comment]
@mo1seev, 

I think we can use #475 for this.

**Comment by mo1seev** [comment]
@ollo69 im not sure, that its same, but lets wait and try =)

**Comment by mo1seev** [comment]
Bcs this device also have Ionizer as on AC. And its work for AC and not work for this device.

AC:
<img width="347" alt="image" src="https://user-images.githubusercontent.com/23101489/225593230-20bd0d43-2335-4dc1-83fc-735a70c68555.png">

Purifier:
<img width="695" alt="image" src="https://user-images.githubusercontent.com/23101489/225593669-eda7607a-07aa-4fa0-9824-9f4a7511ee7c.png">
